pkg/flow/logging/level/level.go

```diff
@@ -0,0 +1,50 @@
+package level
+
+import (
+	"context"
+	"log/slog"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/agent/pkg/flow/logging"
+)
+
+const (
+	levelKey = "level"
+)
+
+// Error returns a logger that includes a Key/ErrorValue pair.
+func Error(logger log.Logger) log.Logger {
+	return toLevel(logger, "error", slog.LevelError)
+}
+
+// Warn returns a logger that includes a Key/WarnValue pair.
+func Warn(logger log.Logger) log.Logger {
+	return toLevel(logger, "warn", slog.LevelWarn)
+}
+
+// Info returns a logger that includes a Key/InfoValue pair.
+func Info(logger log.Logger) log.Logger {
+	return toLevel(logger, "info", slog.LevelInfo)
+}
+
+// Debug returns a logger that includes a Key/DebugValue pair.
+func Debug(logger log.Logger) log.Logger {
+	return toLevel(logger, "debug", slog.LevelDebug)
+}
+
+func toLevel(logger log.Logger, level string, slogLevel slog.Level) log.Logger {
+	if adapter, ok := logger.(logging.EnabledAware); ok {
+		if !adapter.Enabled(context.Background(), slogLevel) {
+			return disabledLogger
```

Comment:
The edge case here is that: `level.Error(level.Debug(logger))` (i.e. setting level twice) will not work. I don't think that should be a problem given how we use the loggers right now.